nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/trash/TrashService.java

```diff
@@ -178,6 +186,17 @@ default void untrashDocument(DocumentModel doc) {
      */
     String mangleName(DocumentModel doc);
 
+    /**
+     * Unmangles the name of a document in the trash to find its un-trashed name.
+     *
+     * @param session the session used for collisions check
+     * @param docName the document name
+     * @param parentRef the parent document ref
+     * @return the unmangled name
+     * @since 11.1
+     */
+    String unmangleName(CoreSession session, String docName, DocumentRef parentRef);
```

Comment:
I would go for the same parameters order than `CoreSession#move`: session, parentRef and docName. wdyt?

Can you provide a default implementation which throws an exception as it's a HF?